### ISSUE TITLE
Pillar no longer obeys client set environment in 2015.8

### ISSUE THREAD

**Issue by dferrantelli** [saltstack/salt#28237]
Here is a sample scenario:

top.sls
```salt
prod:
  '*':
    - base_users
    - base_pkgs
  'app*':
    - webapp
staging:
  '*':
    - base_users
    - base_pkgs
  'app*':
    - webapp
dev:
  '*':
    - base_users
    - base_pkgs
    - dev_users
  'app*':
    - webapp
    - rabbitmq
```

Given the above pillar top file, every minion renders the `base_users` and `base_packages` file three times - even when the `environment` is set in the minion config. Every non dev minion would also get the `dev_users` data in this scenario as well.

**Comment by basepi** [comment]
What version was this last working in for you? I've heard reports of this (in fact, I'm pretty sure this is a duplicate issue, just don't have the original in front of me), and I'm trying to track down where exactly it broke.

**Comment by dferrantelli** [comment]
The last version it reliably worked for me was 2015.5.3. I had missed a few versions, so the change is not definitely between 2015.5.3 and 2015.5.4. I tried a quick search, sorry for the duplicate issue if it is one.

**Comment by basepi** [comment]
Thanks, @dferrantelli, we'll look into this.

**Comment by DmitryKuzmenko** [comment]
@dferrantelli the cause of the changed behavior is 0eedf7ca2 that is part of #23926. It was initially released in 2015.8.0
The change was intended to split ```pillarenv``` from ```saltenv``` because previously it was mixed in some places and ```pillarenv``` was unable to be set in some.
Now to set pillar environment in minion config you have to use ```pillarenv``` configuration key instead of ```environment```.

**Comment by DmitryKuzmenko** [comment]
There is more than a month of silence here. The provided information should help so I'm going to close this.
